replica/database.cc

```diff
@@ -2833,29 +2800,70 @@ future<> database::drain() {
     b.cancel();
 }
 
+void database::tables_metadata::add_table_helper(database& db, keyspace& ks, table& cf, schema_ptr s) {
+    // A table needs to be added atomically.
+    auto id = s->id();
+    ks.add_or_update_column_family(s);
+    auto remove_cf1 = defer([&] () noexcept { ks.metadata()->remove_column_family(s); });
+    s->registry_entry()->set_table(cf.weak_from_this());
+
```

Comment:
Doesn't this need an undo too?

Reply:
Or maybe the registry is able to auto-prune removed tables via the weak pointer? If so a comment is in order.